command/results_gotest.go

```diff
@@ -121,33 +121,29 @@ func sendTestLogsAndResults(ctx context.Context, comm client.Communicator, logge
 	return nil
 }
 
-// AllOutputFiles creates a list of all test output files that will be parsed, by expanding
-// all of the file patterns specified to the command.
-func (c *goTestResults) allOutputFiles() ([]string, error) {
+// globFiles returns a unique set of files that match the given glob patterns.
+func globFiles(patterns ...string) ([]string, error) {
+	var matchedFiles []string
 
-	outputFiles := []string{}
-
-	// walk through all specified file patterns
-	for _, pattern := range c.Files {
+	for _, pattern := range patterns {
 		matches, err := filepath.Glob(pattern)
 		if err != nil {
-			return nil, errors.Wrap(err, "error expanding file patterns")
+			return nil, errors.Wrapf(err, "globbing file pattern '%s'", pattern)
 		}
-		outputFiles = append(outputFiles, matches...)
+		matchedFiles = append(matchedFiles, matches...)
 	}
 
-	// uniquify the list
-	asSet := map[string]bool{}
-	for _, file := range outputFiles {
-		asSet[file] = true
+	// Uniquify the matches
+	asSet := map[string]struct{}{}
```

Comment:
What was the reason for changing this? map[string]bool is a standard way of defining sets. struct{} seems to unnecessarily complicate it.

Reply:
Using `map[T]struct{}{}` is often used to indicate that it's a set and the value is irrelevant. The code base uses both `map[T]bool` and `map[T]struct{}` for sts so I think it's fine as is.

Reply:
I think map[string]bool is more readable: it reads as "does this exist, true or false" whereas map[string]struct{}{} may implies the value has some data. The go community seems torn but I guess I don't see a reason to make this change and mess with the git blame when it's fine as it is.

Reply:
That seems valid. I changed it back to map[string]bool.